### DISCUSSION TITLE
pushImage from SPIFFS

### DISCUSSION THREAD

**Discussion by peppdiy** [Bodmer/TFT_eSPI#1430] [Q&A - General]
On a ESP32 is possible to store bitmap images data to use with pushImage function in SPIFFS or FAT instead of flash?

**Comment by Bodmer** [comment]
The TFT_SPIFS_BMP example displays BMP files stored in SPIFFS.  Bitmaps on SD card can also be read.

pushImage can only be used once the image is in RAM (or PSRAM for ESP32).

**Comment by peppdiy** [comment]
Thank you, I tested the BMP example and works, but SPIFFS is a bit slow and I've checked that with LittleFS is much better

pushImage is faster but the space in flash is limited

**Reply by Bodmer** [reply]
SPIFFS and LittleFS use Flash so have no space advantage over a Flash array.

**Comment by Bodmer** [comment]
You can use pushImage to copy a FLASH array direct to the TFT, but I guess you realise that from the examples.
